### ISSUE TITLE
Aewan ASCII Editor

### ISSUE THREAD

**Issue by Sussy-OS** [Botspot/pi-apps#1515]
### What is the name of the app?

Aewan

### (Optional) Where is the app hosted?

http://aewan.sourceforge.net/

### About the app

Aewan is a curses-based program that allows for the creation and editing of ascii art. The user is able to move the cursor around the screen by means of the arrow keys and 'paint' characters by pressing the corresponding keys. There are dialog boxes that allow the user to choose foreground and background colors, as well as bold and blink attributes. The user may also select rectangular areas of the canvas in order to move, copy and paste them. Aewan also supports 'intelligent' horizontal and vertical flipping (e.g. converts '\' to '/', etc).

What sets Aewan apart from similar projects is the fact that it can work with multiple layers, and has the ability to turn transparency and visibility on and off for each layer. A layer dialog is provided through which the user can change the order of the layers. Thus, each layer can be edited independently in order to generate a composite drawing. Instead of using the layers for compositing, it is also possible to use the layers as frames for an animation, thus enabling the user to create ascii animations with Aewan.

Aewan can export animations to a "less movie", that is, a regular file that will show the animation when when paged through in the 'less' pager or similar program (even Notepad will work if you don't need color). It can also export a shellscript which will play the animation on the terminal when executed.

The file format is easy to parse, so it is easy to write a terminal-based application that uses the Aewan files to display onscreen. Currently it has been tested on the Linux terminal, rxvt, xterm, the Cygwin terminal and the FreeBSD console.

### Upload file

[Aewan Editor.zip](https://github.com/Botspot/pi-apps/files/8118913/Aewan.Editor.zip)

### Confirmations

- [X] I have confirmed that this app is legal and not piracy.
- [x] I have confirmed that this app has never been discussed in https://github.com/Botspot/pi-apps/issues and https://github.com/Botspot/pi-apps/pulls, and it is not in the Pi Apps app list.
- [X] I have confirmed that this app can run on Raspberry Pi.

**Comment by Sussy-OS** [comment]
@Botspot If you give the `A-Okay` then we can move this to a pull request.

**Comment by Sussy-OS** [comment]
*Just edited .zip now.*

**Comment by Sussy-OS** [comment]
btw, to convert to ASCII, use the `aecat -o mypicture.ansi mypicture.ae` command.

**Comment by Botspot** [comment]
Very cool. So far, I see no reason *not* to add it. Maybe add a menu button.

I tried to get this running. It says "[Document contains no layers; press F1 for menu]"
I press F1, the word "New" is highlighted, I press Enter, but nothing happens. I also tried typing "n". What am I missing?

**Comment by Sussy-OS** [comment]
> Very cool. So far, I see no reason _not_ to add it. Maybe add a menu button.
> 
> I tried to get this running. It says "[Document contains no layers; press F1 for menu]" I press F1, the word "New" is highlighted, I press Enter, but nothing happens. I also tried typing "n". What am I missing?

@Botspot Ah, so this program is layer-based, so you need to create a new layer by pressing Ctrl+A and *then* select a color by pressing F2 then C, and to move the cursor around, use the arrow keys, and letters/numbers/sybols to type.

**Comment by Sussy-OS** [comment]
We *should* probably add a menu button, but I have one question: what terminal should I default to?

**Comment by Sussy-OS** [comment]
And- one more thing: to open the help menu, perss F4, then K

**Comment by Sussy-OS** [comment]
*Edited description*

**Comment by Itai-Nelken** [comment]
> We _should_ probably add a menu button, but I have one question: what terminal should I default to?

Use pi-apps's `terminal-run` script, it will choose an available terminal out of the ones pi-apps currently supports.

**Comment by Botspot** [comment]
> > We _should_ probably add a menu button, but I have one question: what terminal should I default to?
> 
> Use pi-apps's `terminal-run` script, it will choose an available terminal out of the ones pi-apps currently supports.

Like this: (snatched from zoom script)
```bash
echo "Creating a Zoom button in the Main Menu..."
echo "[Desktop Entry]
Version=${version}
Name=Zoom
Exec=${DIRECTORY}/etc/terminal-run "\""$HOME/zoom/runzoom.sh %u"\"" 'Close this window to exit Zoom'
Icon=$(dirname "$0")/icon-64.png
Path=${HOME}/zoom/
Type=Application
Comment=i386 version of software platform used for teleconferencing using Box86
Categories=Network;
StartupNotify=true" > ~/.local/share/applications/zoom.desktop
```

**Comment by Sussy-OS** [comment]
Okay, I just finished it, and I edited the original .zip file, but if you don't wait to scroll all the way back up to the top, here it is: 
[Aewan Editor.zip](https://github.com/Botspot/pi-apps/files/8118907/Aewan.Editor.zip)

May we move this to a pull request, if all is well?

**Comment by Itai-Nelken** [comment]
> When using Pi-Apps terminal plugins (either one) to install/uninstall/reinstall this app, you will need to type `Aewan\ Editor` instead of `Aewan Editor`

That is incorrect with the `bash` plugin. there was a bug that made it not work, but I just fixed it.

**Comment by Sussy-OS** [comment]
> > When using Pi-Apps terminal plugins (either one) to install/uninstall/reinstall this app, you will need to type `Aewan\ Editor` instead of `Aewan Editor`
> 
> That is incorrect with the `bash` plugin. there was a bug that made it not work, but I just fixed it.

Ah, I did not know that. And, the bug happens with `Nano` editor too.

**Comment by Sussy-OS** [comment]
Okay, *now* is this ready to be moved to a pull request? (I am going to make one more change before we do so.)

**Comment by Sussy-OS** [comment]
> One question about usability: @Sussy-OS, if one was sufficiently trained, would they be able to create the colorized Pi-Apps logo in this editor? Or does it strictly ONLY support ASCII characters, and nothing more?

I think you *may* be able to. Would you like me to check?

**Comment by Botspot** [comment]
> > One question about usability: @Sussy-OS, if one was sufficiently trained, would they be able to create the colorized Pi-Apps logo in this editor? Or does it strictly ONLY support ASCII characters, and nothing more?
> 
> I think you _may_ be able to. Would you like me to check?

I think that fewer people will see the point if it was limited to the characters found on an average keyboard.

Basically, in the description, I'd like to say that this tool can be used to create the Pi-Apps in-terminal logo. If that statement could be made, many people would recognize its usefulness. But we can only make that claim if it's true. :)

**Comment by Sussy-OS** [comment]
> > > One question about usability: @Sussy-OS, if one was sufficiently trained, would they be able to create the colorized Pi-Apps logo in this editor? Or does it strictly ONLY support ASCII characters, and nothing more?
> > 
> > 
> > I think you _may_ be able to. Would you like me to check?
> 
> I think that fewer people will see the point if it was limited to the characters found on an average keyboard.
> 
> Basically, in the description, I'd like to say that this tool can be used to create the Pi-Apps in-terminal logo. If that statement could be made, many people would recognize its usefulness. But we can only make that claim if it's true. :)

Okay, let me check..

**Comment by Sussy-OS** [comment]
I *am* able to get it to work, besides the slanted lines.

**Comment by Sussy-OS** [comment]
Oops, did not mean to close it

**Comment by Botspot** [comment]
> I _am_ able to get it to work, besides the slanted lines.

Screenshot?

**Comment by Sussy-OS** [comment]
> > I _am_ able to get it to work, besides the slanted lines.
> 
> Screenshot?

Just the logo, or the text too?

**Comment by Botspot** [comment]
> Just the logo, or the text too?

If you could create both, then show me both.

**Comment by Sussy-OS** [comment]
Okay. Already did, just resizing the logo, as I originally made it the incorrect size.

**Comment by Sussy-OS** [comment]
![image](https://user-images.githubusercontent.com/98842253/155212271-9d19aa46-2440-4f68-ac4b-c5c19433d2b1.png)

**Comment by Sussy-OS** [comment]
If I were to resize it a few more down, and a few more to the left/right, it would look like it

**Comment by Sussy-OS** [comment]
also- it saved the file to ~/pi-apps/etc/ is there any way to fix that? (Because the path is set to ~/pi-apps/ect/ but I dont know if I can change it.)

**Comment by Botspot** [comment]
> also- it saved the file to ~/pi-apps/etc/ is there any way to fix that? (Because the path is set to ~/pi-apps/ect/ but I dont know if I can change it.)

I'm mostly interested if this ASCII editor can support all of the specialty angled Unicode characters. Building the Pi-Apps logo out of full blocks doesn't really answer my question.
Basically, I'm curious if Aewam can add and manipulate the characters found on this page: https://unicode-search.net/unicode-namesearch.pl?term=block&.submit=Search

**Comment by Sussy-OS** [comment]
The official one will never get this update, it is no longer maintained. I am going to fork the sourceforge and see what I can do.

**Comment by Sussy-OS** [comment]
May take a bit longer to get back to you than I thought..
![image](https://user-images.githubusercontent.com/98842253/155611052-3a7adca8-3790-43db-9e08-1927d3b40771.png)

I will try to duplicate the choosing method of the color selector, but instead, edit it to select sybols, instead

**Comment by theofficialgman** [comment]
Are there any more modern alternatives to this application that are not 2 decades old? This application likely isn't aware of any UNICODE changes in the past 20 years as mentioned here https://github.com/Botspot/pi-apps/issues/1515#issuecomment-1049344490

**Comment by matu6968** [comment]
> Are there any more modern alternatives to this application that are not 2 decades old? This application likely isn't aware of any UNICODE changes in the past 20 years as mentioned here [#1515 (comment)](https://github.com/Botspot/pi-apps/issues/1515#issuecomment-1049344490)

One could be [durdraw](https://github.com/cmang/durdraw) but it requires Python =>3.10 which is unsuitable for bullseye since it has Python 3.9
